sql/src/main/java/io/druid/sql/avatica/DruidStatement.java

```diff
@@ -94,7 +103,7 @@ public DruidStatement(
     this.statementId = statementId;
     this.queryContext = queryContext == null ? ImmutableMap.of() : queryContext;
     this.onClose = Preconditions.checkNotNull(onClose, "onClose");
-    this.yielderOpenCloseExecutor = Executors.newSingleThreadExecutor();
+    this.yielderOpenCloseExecutor = Execs.singleThreaded("JDBCYielderOpenCloseExecutor");
```

Comment:
Added a statementId to the executor name and a call to shutdownNow() in DruidStatement.close()